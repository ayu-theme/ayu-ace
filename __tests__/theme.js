import fs from 'fs'
import path from 'path'

test('themes', () => {
	const dark = fs.readFileSync(path.resolve('dark.js'), 'utf-8')
	const light = fs.readFileSync(path.resolve('light.js'), 'utf-8')
	const mirage = fs.readFileSync(path.resolve('mirage.js'), 'utf-8')

	expect(dark).toMatchSnapshot()
	expect(light).toMatchSnapshot()
	expect(mirage).toMatchSnapshot()
})
