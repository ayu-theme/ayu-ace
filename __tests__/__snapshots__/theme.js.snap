exports[`test themes 1`] = `
"/*!
* MIT License
*
* Copyright (c) 2016 Ike Ku
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the \"Software\"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/
ace.define(\'ace/theme/ayu-dark\', [\'require\', \'exports\', \'module\', \'ace/lib/dom\'], function (acequire, exports, module) {
exports.isDark = false
exports.cssClass = \'ace-ayu-dark\'
exports.cssText = \'.ace-ayu-dark .ace_gutter {\\
background: #0F1419;\\
color: rgb(123,123,116)\\
}\\
\\
.ace-ayu-dark .ace_print-margin {\\
width: 1px;\\
background: #e8e8e8\\
}\\
\\
.ace-ayu-dark {\\
background-color: #0F1419;\\
color: #E6E1CF\\
}\\
\\
.ace-ayu-dark .ace_cursor {\\
color: #F29718\\
}\\
\\
.ace-ayu-dark .ace_marker-layer .ace_selection {\\
background: #253340\\
}\\
\\
.ace-ayu-dark.ace_multiselect .ace_selection.ace_start {\\
box-shadow: 0 0 3px 0px #0F1419;\\
border-radius: 2px\\
}\\
\\
.ace-ayu-dark .ace_marker-layer .ace_step {\\
background: rgb(198, 219, 174)\\
}\\
\\
.ace-ayu-dark .ace_marker-layer .ace_bracket {\\
margin: -1px 0 0 -1px;\\
border: 1px solid #FF0000\\
}\\
\\
.ace-ayu-dark .ace_marker-layer .ace_active-line {\\
background: #151A1F\\
}\\
\\
.ace-ayu-dark .ace_gutter-active-line {\\
background-color: #151A1F\\
}\\
\\
.ace-ayu-dark .ace_marker-layer .ace_selected-word {\\
border: 1px solid #253340\\
}\\
\\
.ace-ayu-dark .ace_fold {\\
background-color: #FFB454;\\
border-color: #E6E1CF\\
}\\
\\
.ace-ayu-dark .ace_keyword {\\
color: #FF7733\\
}\\
\\
.ace-ayu-dark .ace_keyword.ace_other.ace_unit {\\
color: #FFEE99\\
}\\
\\
.ace-ayu-dark .ace_constant.ace_language {\\
color: #FFEE99\\
}\\
\\
.ace-ayu-dark .ace_constant.ace_numeric {\\
color: #FFEE99\\
}\\
\\
.ace-ayu-dark .ace_constant.ace_character {\\
color: #FFEE99\\
}\\
\\
.ace-ayu-dark .ace_constant.ace_character.ace_escape {\\
color: #95E6CB\\
}\\
\\
.ace-ayu-dark .ace_support.ace_function {\\
color: #FFB454\\
}\\
\\
.ace-ayu-dark .ace_support.ace_constant {\\
color: #FFEE99\\
}\\
\\
.ace-ayu-dark .ace_support.ace_class {\\
color: #36A3D9\\
}\\
\\
.ace-ayu-dark .ace_support.ace_type {\\
color: #36A3D9\\
}\\
\\
.ace-ayu-dark .ace_storage.ace_type {\\
color: #FF7733\\
}\\
\\
.ace-ayu-dark .ace_invalid {\\
color: #FF3333\\
}\\
\\
.ace-ayu-dark .ace_invalid.ace_deprecated {\\
color: #FFFFFF;\\
background-color: #FF7733\\
}\\
\\
.ace-ayu-dark .ace_string {\\
color: #B8CC52\\
}\\
\\
.ace-ayu-dark .ace_string.ace_regexp {\\
color: #95E6CB\\
}\\
\\
.ace-ayu-dark .ace_comment {\\
font-style: italic;\\
color: #5C6773\\
}\\
\\
.ace-ayu-dark .ace_variable {\\
color: #E6E1CF\\
}\\
\\
.ace-ayu-dark .ace_variable.ace_language {\\
font-style: italic;\\
color: #36A3D9\\
}\\
\\
.ace-ayu-dark .ace_variable.ace_parameter {\\
color: #FFEE99\\
}\\
\\
.ace-ayu-dark .ace_meta.ace_tag {\\
color: #E7C547\\
}\\
\\
.ace-ayu-dark .ace_entity.ace_other.ace_attribute-name {\\
color: #FF7733\\
}\\
\\
.ace-ayu-dark .ace_entity.ace_name.ace_function {\\
color: #FFB454\\
}\\
\\
.ace-ayu-dark .ace_entity.ace_name.ace_tag {\\
color: #36A3D9\\
}\\
\\
.ace-ayu-dark .ace_markup.ace_heading {\\
color: #B8CC52\\
}\'

var dom = acequire(\'../lib/dom\')
dom.importCssString(exports.cssText, exports.cssClass)
})"
`;

exports[`test themes 2`] = `
"/*!
* MIT License
*
* Copyright (c) 2016 Ike Ku
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the \"Software\"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/
ace.define(\'ace/theme/ayu-light\', [\'require\', \'exports\', \'module\', \'ace/lib/dom\'], function (acequire, exports, module) {
exports.isDark = false
exports.cssClass = \'ace-ayu-light\'
exports.cssText = \'.ace-ayu-light .ace_gutter {\\
background: #FAFAFA;\\
color: rgb(171,177,183)\\
}\\
\\
.ace-ayu-light .ace_print-margin {\\
width: 1px;\\
background: #e8e8e8\\
}\\
\\
.ace-ayu-light {\\
background-color: #FAFAFA;\\
color: #5C6773\\
}\\
\\
.ace-ayu-light .ace_cursor {\\
color: #FF6A00\\
}\\
\\
.ace-ayu-light .ace_marker-layer .ace_selection {\\
background: #F0EEE4\\
}\\
\\
.ace-ayu-light.ace_multiselect .ace_selection.ace_start {\\
box-shadow: 0 0 3px 0px #FAFAFA;\\
border-radius: 2px\\
}\\
\\
.ace-ayu-light .ace_marker-layer .ace_step {\\
background: rgb(198, 219, 174)\\
}\\
\\
.ace-ayu-light .ace_marker-layer .ace_bracket {\\
margin: -1px 0 0 -1px;\\
border: 1px solid #FF0000\\
}\\
\\
.ace-ayu-light .ace_marker-layer .ace_active-line {\\
background: #F2F2F2\\
}\\
\\
.ace-ayu-light .ace_gutter-active-line {\\
background-color: #F2F2F2\\
}\\
\\
.ace-ayu-light .ace_marker-layer .ace_selected-word {\\
border: 1px solid #F0EEE4\\
}\\
\\
.ace-ayu-light .ace_fold {\\
background-color: #F29718;\\
border-color: #5C6773\\
}\\
\\
.ace-ayu-light .ace_keyword {\\
color: #F2590C\\
}\\
\\
.ace-ayu-light .ace_keyword.ace_other.ace_unit {\\
color: #A37ACC\\
}\\
\\
.ace-ayu-light .ace_constant.ace_language {\\
color: #A37ACC\\
}\\
\\
.ace-ayu-light .ace_constant.ace_numeric {\\
color: #A37ACC\\
}\\
\\
.ace-ayu-light .ace_constant.ace_character {\\
color: #A37ACC\\
}\\
\\
.ace-ayu-light .ace_constant.ace_character.ace_escape {\\
color: #4DBF99\\
}\\
\\
.ace-ayu-light .ace_support.ace_function {\\
color: #F29718\\
}\\
\\
.ace-ayu-light .ace_support.ace_constant {\\
color: #A37ACC\\
}\\
\\
.ace-ayu-light .ace_support.ace_class {\\
color: #41A6D9\\
}\\
\\
.ace-ayu-light .ace_support.ace_type {\\
color: #41A6D9\\
}\\
\\
.ace-ayu-light .ace_storage.ace_type {\\
color: #F2590C\\
}\\
\\
.ace-ayu-light .ace_invalid {\\
color: #FF3333\\
}\\
\\
.ace-ayu-light .ace_invalid.ace_deprecated {\\
color: #FFFFFF;\\
background-color: #F2590C\\
}\\
\\
.ace-ayu-light .ace_string {\\
color: #86B300\\
}\\
\\
.ace-ayu-light .ace_string.ace_regexp {\\
color: #4DBF99\\
}\\
\\
.ace-ayu-light .ace_comment {\\
font-style: italic;\\
color: #ABB0B6\\
}\\
\\
.ace-ayu-light .ace_variable {\\
color: #5C6773\\
}\\
\\
.ace-ayu-light .ace_variable.ace_language {\\
font-style: italic;\\
color: #41A6D9\\
}\\
\\
.ace-ayu-light .ace_variable.ace_parameter {\\
color: #A37ACC\\
}\\
\\
.ace-ayu-light .ace_meta.ace_tag {\\
color: #E7C547\\
}\\
\\
.ace-ayu-light .ace_entity.ace_other.ace_attribute-name {\\
color: #F2590C\\
}\\
\\
.ace-ayu-light .ace_entity.ace_name.ace_function {\\
color: #F29718\\
}\\
\\
.ace-ayu-light .ace_entity.ace_name.ace_tag {\\
color: #41A6D9\\
}\\
\\
.ace-ayu-light .ace_markup.ace_heading {\\
color: #86B300\\
}\'

var dom = acequire(\'../lib/dom\')
dom.importCssString(exports.cssText, exports.cssClass)
})"
`;

exports[`test themes 3`] = `
"/*!
* MIT License
*
* Copyright (c) 2016 Ike Ku
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the \"Software\"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/
ace.define(\'ace/theme/ayu-mirage\', [\'require\', \'exports\', \'module\', \'ace/lib/dom\'], function (acequire, exports, module) {
exports.isDark = false
exports.cssClass = \'ace-ayu-mirage\'
exports.cssText = \'.ace-ayu-mirage .ace_gutter {\\
background: #212733;\\
color: rgb(125,127,129)\\
}\\
\\
.ace-ayu-mirage .ace_print-margin {\\
width: 1px;\\
background: #e8e8e8\\
}\\
\\
.ace-ayu-mirage {\\
background-color: #212733;\\
color: #D9D7CE\\
}\\
\\
.ace-ayu-mirage .ace_cursor {\\
color: #FFCC66\\
}\\
\\
.ace-ayu-mirage .ace_marker-layer .ace_selection {\\
background: #343F4C\\
}\\
\\
.ace-ayu-mirage.ace_multiselect .ace_selection.ace_start {\\
box-shadow: 0 0 3px 0px #212733;\\
border-radius: 2px\\
}\\
\\
.ace-ayu-mirage .ace_marker-layer .ace_step {\\
background: rgb(198, 219, 174)\\
}\\
\\
.ace-ayu-mirage .ace_marker-layer .ace_bracket {\\
margin: -1px 0 0 -1px;\\
border: 1px solid #FF0000\\
}\\
\\
.ace-ayu-mirage .ace_marker-layer .ace_active-line {\\
background: #242B38\\
}\\
\\
.ace-ayu-mirage .ace_gutter-active-line {\\
background-color: #242B38\\
}\\
\\
.ace-ayu-mirage .ace_marker-layer .ace_selected-word {\\
border: 1px solid #343F4C\\
}\\
\\
.ace-ayu-mirage .ace_fold {\\
background-color: #FFD580;\\
border-color: #D9D7CE\\
}\\
\\
.ace-ayu-mirage .ace_keyword {\\
color: #FFAE57\\
}\\
\\
.ace-ayu-mirage .ace_keyword.ace_other.ace_unit {\\
color: #D4BFFF\\
}\\
\\
.ace-ayu-mirage .ace_constant.ace_language {\\
color: #D4BFFF\\
}\\
\\
.ace-ayu-mirage .ace_constant.ace_numeric {\\
color: #D4BFFF\\
}\\
\\
.ace-ayu-mirage .ace_constant.ace_character {\\
color: #D4BFFF\\
}\\
\\
.ace-ayu-mirage .ace_constant.ace_character.ace_escape {\\
color: #95E6CB\\
}\\
\\
.ace-ayu-mirage .ace_support.ace_function {\\
color: #FFD580\\
}\\
\\
.ace-ayu-mirage .ace_support.ace_constant {\\
color: #D4BFFF\\
}\\
\\
.ace-ayu-mirage .ace_support.ace_class {\\
color: #5CCFE6\\
}\\
\\
.ace-ayu-mirage .ace_support.ace_type {\\
color: #5CCFE6\\
}\\
\\
.ace-ayu-mirage .ace_storage.ace_type {\\
color: #FFAE57\\
}\\
\\
.ace-ayu-mirage .ace_invalid {\\
color: #FF3333\\
}\\
\\
.ace-ayu-mirage .ace_invalid.ace_deprecated {\\
color: #FFFFFF;\\
background-color: #FFAE57\\
}\\
\\
.ace-ayu-mirage .ace_string {\\
color: #BAE67E\\
}\\
\\
.ace-ayu-mirage .ace_string.ace_regexp {\\
color: #95E6CB\\
}\\
\\
.ace-ayu-mirage .ace_comment {\\
font-style: italic;\\
color: #5C6773\\
}\\
\\
.ace-ayu-mirage .ace_variable {\\
color: #D9D7CE\\
}\\
\\
.ace-ayu-mirage .ace_variable.ace_language {\\
font-style: italic;\\
color: #5CCFE6\\
}\\
\\
.ace-ayu-mirage .ace_variable.ace_parameter {\\
color: #D4BFFF\\
}\\
\\
.ace-ayu-mirage .ace_meta.ace_tag {\\
color: #80D4FF\\
}\\
\\
.ace-ayu-mirage .ace_entity.ace_other.ace_attribute-name {\\
color: #FFAE57\\
}\\
\\
.ace-ayu-mirage .ace_entity.ace_name.ace_function {\\
color: #FFD580\\
}\\
\\
.ace-ayu-mirage .ace_entity.ace_name.ace_tag {\\
color: #5CCFE6\\
}\\
\\
.ace-ayu-mirage .ace_markup.ace_heading {\\
color: #BAE67E\\
}\'

var dom = acequire(\'../lib/dom\')
dom.importCssString(exports.cssText, exports.cssClass)
})"
`;
